tests/translate/translate_updatedzd.py

```diff
@@ -24,11 +24,19 @@ def __init__(self, grid):
             self.out_vars[v] = self.in_vars["data_vars"][v]
         self.out_vars["wsd"]["kstart"] = grid.npz
         self.out_vars["wsd"]["kend"] = None
+        self.updatedzd = fv3core.stencils.updatedzd.UpdateDeltaZOnDGrid(
+            self.grid, d_sw.get_column_namelist(), d_sw.k_bounds()
+        )
 
     def compute(self, inputs):
         self.make_storage_data_input_vars(inputs)
-        updatedzd = fv3core.stencils.updatedzd.UpdateDeltaZOnDGrid(
-            self.grid, d_sw.get_column_namelist(), d_sw.k_bounds()
-        )
-        updatedzd(**inputs)
-        return self.slice_output(inputs)
+        self.updatedzd(**inputs)
+        self.out_vars["zh"].update(self.updatedzd._zh_validator.translate_dict)
+        outputs = self.slice_output(inputs)
+        outputs["zh"] = self.subset_output("zh", outputs["zh"])
+        return outputs
+
+    def subset_output(self, varname, output):
+        if varname == "zh":
+            output = output[self.updatedzd._zh_validator.validation_slice]
```

Comment:
This works for this case, but want to point out the validation_slice here is defined from the point of reference of the full domain. e.g. because zh starts at isd, jsd, or at the start of the full domain including halos, setting the validation region using 'compute_domain' works just fine. But some data comparisons are of a fields that start at different halo lines, e.g, is-2 or is - 1, and grid.compute_domain() as a slice would be further into the interior than you likely want, all of the familiar grid indexes would be off when slicing the array.

Reply:
Alternatively I wonder if it would make sense to fill the reference data with nans rather than slicing, but slicing is probably easier to debug

Reply:
Filling "not this slice" with nans is also a fairly complicated operation, since there's no way to "invert" a slice.

The impact of this I think is that `validation_slice` has to be the slice on the reference data. If the user sets that in the class, then the test data passed here will already be made to be the reference data shape by `self.slice_output`.

Ah but it will break/require re-writing the test because the NaNs on the array we use in the model should no longer be applied in the same way.

Maybe for now we say this object only works for reference data with 3 halo points, and we can fix this if we run into this issue? I think it can be done by adding a new keyword argument about the validation origin and then changing the tests and the code inside the validation class. But it's also possible we won't need it for that use case.